collect_app/src/main/java/org/odk/collect/android/formmanagement/formmap/FormMapViewModel.kt

```diff
@@ -149,11 +149,12 @@ class FormMapViewModel(
                 info
             )
         } else {
-            val action = if (instance.status == Instance.STATUS_INCOMPLETE) {
-                createEditAction()
-            } else {
-                createViewAction()
-            }
+            val action =
+                if (instance.status == Instance.STATUS_INCOMPLETE || instance.status == Instance.STATUS_VALID) {
```

Comment:
Shouldn't  I`nstance.STATUS_INVALID` be here too?

Reply:
Ooops! Yeah it should.

Reply:
I've pulled out a shared helper for this so we don't have to repeat/test this logic in multiple places.